static/js/message_list_view.js

```diff
@@ -162,12 +162,25 @@ MessageListView.prototype = {
             }
         }
 
+        function is_same_day(last_message, this_message) {
+            var date = new Date(last_message.timestamp * 1000);
+            var today = new Date(this_message.timestamp * 1000);
+
+            return (date.getDate() === today.getDate() &&
+                date.getMonth() === today.getMonth() &&
+                date.getFullYear() === today.getFullYear());
+        }
+
         _.each(message_containers, function (message_container) {
             message_container.include_recipient = false;
             message_container.include_footer    = false;
 
+            var msg_list = message_list.all._items;
             if (!same_day(prev, message_container)) {
-                message_container.print_date = true;
+
+                if (!is_same_day(msg_list[msg_list.length - 1], message_container.msg)) {
+                    message_container.print_date = true;
+                }
```

Comment:
I don't think that this logic is correct in the event that we're rerendering a message not at the end of the feed (e.g. from message editing or scrolling up); this should be comparing to the immediately previous message group, not the very last one.

I also notice we have an existing `message_group.show_date` field; can we just use that for this logic?  

If not, I think it at least shows the correct way to do what we're trying to do here, namely clearing the flag here when merging message groups.
```
                if (same_day(last_msg_container, first_msg_container)) {                                        
                    // Clear the date if it is the same as the last group                                       
                    second_group.show_date = undefined;                                                         
                }                                                                                               
```